### PR TITLE
PERF: reduce usages of `existsAfterExpansion` in annotators

### PR DESCRIPTION
**Pull request by vlad20012**
changelog: 1. Slightly speed up highlighting 2. Speed up appearance of inlay hints

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/annotator/RsCfgDisabledCodeAnnotator.kt [hunk 1]
```diff
@@ -16,11 +16,12 @@ class RsCfgDisabledCodeAnnotator : AnnotatorBase() {
...
         if (holder.isBatchMode) return
 
-        if (element is RsDocAndAttributeOwner && !element.isEnabledByCfgSelf) {
+        val crate = holder.currentAnnotationSession.currentCrate()
+        if (element is RsDocAndAttributeOwner && !element.isEnabledByCfgSelfOrInAttrProcMacroBody(crate)) {
             holder.createCondDisabledAnnotation()
         }
 
-        if (element is RsAttr && element.isDisabledCfgAttrAttribute && element.owner?.isEnabledByCfgSelf == true) {
+        if (element is RsAttr && element.isDisabledCfgAttrAttribute && element.owner?.isEnabledByCfgSelfOrInAttrProcMacroBody == true) {
```
Comment (dima74) [inline src/main/kotlin/org/rust/ide/annotator/RsCfgDisabledCodeAnnotator.kt:24]:
```suggestion
        if (element is RsAttr && element.isDisabledCfgAttrAttribute && element.owner?.isEnabledByCfgSelfOrInAttrProcMacroBody(crate) == true) {
```

File: src/main/kotlin/org/rust/ide/annotator/AnnotationSessionEx.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import com.intellij.openapi.util.Key
+import com.intellij.util.containers.orNull
+import org.rust.lang.core.crate.Crate
+import org.rust.lang.core.psi.RsFile
+import org.rust.openapiext.getOrPut
+import java.util.*
+
+private val CRATE_KEY = Key<Optional<Crate>>("org.rust.ide.annotator.currentCrate")
+
+fun AnnotationSession.currentCrate(): Crate? {
```
Comment (dima74) [inline src/main/kotlin/org/rust/ide/annotator/AnnotationSessionEx.kt:18]: It is a bit long to write `holder.currentAnnotationSession.currentCrate()`, maybe also add helper method?
```kotlin
fun AnnotationHolder.currentCrate(): Crate? = currentAnnotationSession.currentCrate()
```

**Review by dima74 — APPROVED** [review]

**Comment by vlad20012** [comment]
bors r=dima74

**Comment by vlad20012** [comment]
bors r=dima74

**Pull request merged**